src/v/cluster/rm_stm_types.h

```diff
@@ -242,10 +360,29 @@ struct async_adl<tx_snapshot_v4> {
     ss::future<tx_snapshot_v4> from(iobuf_parser&);
 };
 
-using tx_snapshot = cluster::tx::tx_snapshot;
+using tx_snapshot = cluster::tx::tx_snapshot_v5;
```

Comment:
The fact that we have `tx_snapshot` in different namespaces mean different things looks potentially confusing... Can we just use `tx_snapshot_v5` for ADL everywhere?